oem/firstvoices/android/app/src/main/java/com/firstvoices/keyboards/FVShared.java

```diff
@@ -124,53 +129,83 @@ public static FVShared getInstance() {
   }
 
   private FVRegionList loadRegionList() {
-        FVRegionList list = new FVRegionList();
-        try {
-            // At this point in initialization, fv_all.kmp hasn't been extracted, so
-            // we get all the keyboard info from keyboards.csv
-            InputStream inputStream = context.getAssets().open("keyboards.csv");
-            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
-
-            reader.readLine(); // skip header row
-            String line = reader.readLine();
-
-            while (line != null) {
-                while (line.contains(",,"))
-                    line = line.replace(",,", ", ,");
-
-                String[] values = line.split(",");
-                if (values != null && values.length > 0) {
-                    // Read in column info
-                    String kbId = values[1];
-                    String kbName = values[2];
-                    String regionName = values[3];
-                    String legacyId = values[4];
-                    String version = values[5];
-                    String lgId = values[6].toLowerCase(); // Normalize language ID
-                    String lgName = values[7];
-
-                    FVRegion region = list.findRegion(regionName);
-                    if(region == null) {
-                        region = new FVRegion(regionName);
-                        list.add(region);
-                    }
-
-                    FVKeyboard keyboard = new FVKeyboard(kbId, kbName, legacyId, version, lgId, lgName);
-
-                    region.keyboards.add(keyboard);
-                }
+      FVRegionList list = new FVRegionList();
+      File resourceRoot = new File(getResourceRoot());
+      PackageProcessor kmpProcessor = new PackageProcessor(resourceRoot);
+      JSONParser parser = new JSONParser();
+      File jsonFile = new File(getPackagesDir() + FVDefault_PackageID + File.separator + FVKeyboards_JSON);
```

Comment:
If we pass this path in as a parameter, then we could add a unit test for this quite easily I think?

Reply:
I could add the keyboards.json path as a parameter to loadRegionList().

I don't think the FV project is running any unit tests, so I'd have to add some plumbing for that.

Reply:
> Yep, future project :)

Tracked as #12098